parquet-tools/src/main/java/parquet/tools/command/CatCommand.java

```diff
@@ -52,7 +69,11 @@ public void execute(CommandLine options) throws Exception {
       PrintWriter writer = new PrintWriter(Main.out, true);
       reader = new ParquetReader<SimpleRecord>(new Path(input), new SimpleReadSupport());
       for (SimpleRecord value = reader.read(); value != null; value = reader.read()) {
-        value.prettyPrint(writer);
+        if (options.hasOption('j')) {
+          value.prettyPrintJson(writer);
```

Comment:
I think that we should generate the JSON based on the parquet schema (It is passed to the SimpleReadSupport from the file's metadata) rather than inferring isArray and isMap from the fields in each object.
I would make prettyPrintJson not part of SimpleRecord but instead:
prettyPrintJson(value, schema, writer)
and traverse schema at the same time as value and use schema to decide if it should be represented as a map, an array or a string (OriginalType is List, Map or UTF8).